Zend/zend_exceptions.c

```diff
@@ -821,6 +821,14 @@ static zend_object *zend_throw_exception_zstr(zend_class_entry *exception_ce, ze
 		ZVAL_LONG(&tmp, code);
 		zend_update_property_ex(exception_ce, Z_OBJ(ex), ZSTR_KNOWN(ZEND_STR_CODE), &tmp);
 	}
+	if (EG(exception_filename) != NULL) {
```

Comment:
I don't know if it'd make sense to override the `EG(exception_filename)` but not the `EG(exception_lineno)`, but the implementation is correcty.

It might also be a tiny bit more efficient if both were stored into local variables at the same time, since the compiler can't correctly prove zend_update_property_ex doesn't change `EG(exception_lineno)`